### PR TITLE
feat: Support json ld for create resource

### PR DESCRIPTION
**Pull request by seakayone**
- **add updateValueV2fromJsonLd to ApiComplexV2JsonLdRequestParser**
- **implement newValueVersionIri**
- **fixuup**
- **use json ld for create resource**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [x] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [ ] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### CODE REVIEW

**Review by siers — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala [hunk 1]
```diff
@@ -42,8 +54,142 @@ final case class ApiComplexV2JsonLdRequestParser(
...
+    ZIO.scoped {
+      for {
+        model                  <- ModelOps.fromJsonLd(str)
+        resourceAndIri         <- resourceAndIriOption(model)
+        (resource, resourceIri) = resourceAndIri
+        resourceClassIri       <- resourceClassIri(resource)
+        label                  <- ZIO.fromEither(resource.objectString(Rdfs.Label))
+        project                <- attachedToProject(resource)
+        _ <- ZIO
+               .fail("Resource Iri and project Iri must reference the same project")
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala:76]:
```suggestion
               .fail("Resource IRI and project IRI must reference the same project.")
```

File: webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala [hunk 2]
```diff
@@ -42,8 +54,142 @@ final case class ApiComplexV2JsonLdRequestParser(
...
+    } yield user.getOrElse(requestingUser)
+
+  private def checkUser(requestingUser: User, userIri: UserIri, projectIri: ProjectIri): IO[String, User] =
+    requestingUser match {
+      case _ if requestingUser.id == userIri.value => ZIO.succeed(requestingUser)
+      case _
+          if !(requestingUser.permissions.isSystemAdmin ||
+            requestingUser.permissions.isProjectAdmin(projectIri.value)) =>
+        ZIO.fail(
+          s"You are logged in as ${requestingUser.username}, but only a system administrator or project administrator can perform an operation as another user",
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala:166]:
```suggestion
          s"You are logged in as ${requestingUser.username}, but only a system or project administrator can perform an operation as another user.",
```

**Review by mpro7 — APPROVED** [review]
Nice cleanup, thanks! Didn't I already review part of the changes yesterday?

File: webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/common/ApiComplexV2JsonLdRequestParser.scala]: Thanks.

**Pull request merged**